File: GTiHub/GTiHub/Models/Project.cs
namespace GTiHub.Models.EntityModel
{
    #region

    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;

    #endregion

    public class Project : BaseEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ProjectId { get; set; }
        public string Description { get; set; }
        public string Name { get; set; }
        public string Project_Type { get; set; }

        public  Client Client { get; set; }
        public int ClientId { get; set; }
        public  ICollection<ProjectMap> ProjectMaps { get; set; }
        public  ICollection<ProjectSource> ProjectSources { get; set; }
        public  ICollection<ProjectTarget> ProjectTargets { get; set; }
        public  ICollection<UserProjectSec> UserProjectSecs { get; set; }
    }
}

File: GTiHub/GTiHub/ClientApp/app/components/project/project.ts
import { Client } from "../client/client";

export interface IProject {
    projectId?: number;
    name: string;
    description: string;
    project_Type: string;
    client: Client;
    projectMaps?: any[];
    projectSources?: any[];
    projectTargets?: any[];
    created_By?: string;
    creation_Date?: Date;
    modified_By?: string;
    date_Modified?: Date;
}

export class Project implements IProject {
    constructor(
        public name: string,
        public description: string,
        public project_Type: string,
        public client: Client,
        public projectId?: number,
        public projectMaps?: any[],
        public projectSources?: any[],
        public projectTargets?: any[],
        public created_By?: string,
        public creation_Date?: Date,
        public modified_By?: string,
        public date_Modified?: Date) { }
}